### PR TITLE
Add commit-hash option in `sourcemaps upload` command

### PR DESCRIPTION
**Pull request by takaya1992**
### What and why?

https://github.com/DataDog/datadog-ci/issues/753 override commit hash when `sourcemaps upload`.

### How?

add `--commit-hash` option in `sourcemaps upload` command.

### Review checklist

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration)

### CODE REVIEW

File: src/commands/sourcemaps/README.md [hunk 1]
```diff
@@ -69,6 +69,14 @@ The value can be overridden with `--repository-url`.
 Example: With a remote `git@github.com:Datadog/example.git`, links pointing to `https://github.com/Datadog/example` are generated.
 This behavior can be overridden with links to `https://gitlab.com/Datadog/example` with the flag `--repository-url=https://gitlab.com/Datadog/example`.
 
+#### Override commit hash
+
+The value can be overridden with `--commit-hash`.
+
+Example: Pass commit hash when there is no `.git/` directory, such as build in Docker.
```
Comment (maycmlee) [inline src/commands/sourcemaps/README.md:76]:
```suggestion
Example: Pass commit hash when there is no `.git/` directory, such as a build in Docker.
```

**Review by maycmlee — COMMENTED** [review]
Just a small edit, let me know if it changes the meaning.

File: src/commands/sourcemaps/README.md (hunk 1 above)
Comment (takaya1992) [reply src/commands/sourcemaps/README.md]: thanks

**Review by maycmlee — APPROVED** [review]
👍 for docs

**Review by louiszawadzki — APPROVED** [review]
All good for me 👍

**Comment by takaya1992** [comment]
@maycmlee @louiszawadzki 
Thank you for your code review.
It seems that it is necessary to get other members to do a code review as well.
Do you need me to take any action?

File: src/commands/flutter-symbols/upload.ts [hunk 2]
```diff
@@ -194,7 +194,7 @@ export class UploadCommand extends Command {
 
   private async getGitMetadata(): Promise<RepositoryData | undefined> {
     try {
-      return await getRepositoryData(await newSimpleGit(), this.repositoryUrl)
+      return await getRepositoryData(await newSimpleGit(), this.repositoryUrl, undefined)
```
Comment (juan-fernandez) [inline src/commands/flutter-symbols/upload.ts:197]: nitpick: couldn't we have an optional parameter to the function instead of passing `undefined`?

**Review by juan-fernandez — APPROVED** [review]
looks good for CI Visibility. Left a nitpick comment

**Comment by rpelliard** [comment]
Hello @takaya1992,

As I understand it from https://github.com/DataDog/datadog-ci/issues/753, you're trying to get links from the stack trace to your repository, without having access to the `.git` directory during sourcemaps upload, right ?

In order for this feature to work, the sourcemaps upload command needs to collect:
- the repository URL and commit hash
- the list of tracked files in the repository (i.e. output of `git ls-files`, fetched by [`gitTrackedFiles`](https://github.com/DataDog/datadog-ci/blob/5c22d31e4fb6167db93432ebaa4d9dc72dc5c493/src/commands/git-metadata/git.ts#L58))

Without all these, our backend cannot build the links to your repository. This PR looks like a nice way to address the missing commit hash, but the command will still try to retrieve the list of tracked files from the `.git` directory, and will fail.

I'm afraid that these changes will not be enough to solve your problem, and I don't see a simple way to make the metadata upload work without the `.git` directory.

**Comment by dlmr** [comment]
Would it be an alternative to run `datadog-ci git-metadata upload` separately from the `datadog-ci sourcemaps upload` with this flag added? Do you see that as an alternative for getting the tracked files @rpelliard?

I have a similar situation as @takaya1992 and tried to run `git-metadata` separate from the build process where I have the full git context, outside of the Docker container, but it did no replace the need within `sourcemaps`.

**Comment by rpelliard** [comment]
Decoupling the git metadata upload from the sourcemaps upload is the direction we want to go in, but I'm afraid that it won't work "as-is" with the two existing commands (given the way our back-end is made at the moment), and unfortunately there is no simple workaround for now.
I will make sure that this is followed up on internally, and I will close this PR since it won't make the feature work for the use-cases exposed here.

**Pull request closed without merging**